### PR TITLE
Fixes #302 Add status badge for buildstatus of repos

### LINKED ISSUE #302: Implement to show if the documentation is available on a repo (similar to travis)
What kind of buttons could we show on project pages? E.g. could we have a button "yaydoc ok" and "yaydoc failed"?

Please propose and implement a way that projects can include a yaydoc button in their project.

### PR DESCRIPTION
**Pull request by imujjwal96**
## Description

* Build status of a repository is retrieved from the DB and an svg file is generated at the route: `https://yaydoc.herokuapp.com/<owner-name>/<reponame>.svg`
* If a repository is registered but no further commit was made, then yaydoc failed is shown
## Related Issue

issue #302 

## How Has This Been Tested?

Tested. A deployment can be found at https://yaydoc7.herokuapp.com

## Screenshot:
image
image
image

image

## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -27,4 +29,18 @@ router.get('/logs/:type/:email/:uniqueId', function (req, res, next) {
...
+/* Generate SVG image for build status of repository */
+router.get('/:reponame.svg', function (req, res, next) {
+  var reponame = req.params.reponame;
+
+  Repository.getBuildStatusByRepositoryName(reponame, function (error, buildStatus) {
+    if (buildStatus !== undefined) {
+      res.set('Content-Type', 'image/svg+xml');
+      res.render("badge", {
+        status: buildStatus
+      });
```
Comment (sch00lb0y) [inline routes/index.js]: add the else part and handle the error as well.

**Comment by imujjwal96** [comment]
@pri22296  @sch00lb0y  Updated it. The svg for registered repository is generated here: http://yaydoc7.herokuapp.com/imujjwal96/prelimQuiz.svg 
Can be added to a README as here: https://github.com/imujjwal96/prelimQuiz/blob/documentation/README.md
image

**Comment by pri22296** [comment]
@imujjwal96 What happens when the link to svg is not valid?

**Comment by niranjan94** [comment]
> @imujjwal96 What happens when the link to svg is not valid?

@imujjwal96 please show a grey yaydoc invalid svg. And also, ensure the http status code is 404

**Comment by imujjwal96** [comment]
> @imujjwal96 What happens when the link to svg is not valid?

@pri22296 a 404 page was rendered on that link. Didn't check what the status tag would have been in README file.
I'll follow @niranjan94  approach in that case! Thanks for the tip!

**Comment by imujjwal96** [comment]
@pri22296 @niranjan94 Updated.

File: views/badge.jade [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      rect(width="94" height="20" rx="3" fill="#fff")
+    g(clip-path="url(#a)")
+      path(fill="#555" d="M0 0h49v20H0z")
+      path(fill="#9f9f9f" d="M49 0h45v20H49z")
+      path(fill="url(#b)" d="M0 0h94v20H0z")
+    g(fill="#fff" text-anchor="middle" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="11")
+      text(x="24.5" y="15" fill="#010101" fill-opacity=".3") yaydoc
+      text(x="24.5" y="14") yaydoc
+      text(x="70.5" y="15" fill="#010101" fill-opacity=".3") invalid
+      text(x="70.5" y="14") invalid
```
Comment (pri22296) [inline views/badge.jade]: The three blocks are mostly similar. Can something be done to avoid this code repetition?

Comment (imujjwal96) [inline views/badge.jade]: i'll try!

**Comment by imujjwal96** [comment]
@pri22296  Updated.
http://yaydoc7.herokuapp.com/imujjwal96/prelimQudiz.svg -> invalid
http://yaydoc7.herokuapp.com/imujjwal96/prelimQuiz.svg -> success
http://yaydoc7.herokuapp.com/imujjwal96/mydoc.svg -> failed

**Comment by mariobehling** [comment]
@imujjwal96 Clicking on the badge where is the user directed?

**Comment by imujjwal96** [comment]
Currently, we have not documented what to set there. I am.planning to add Travis like build logs in Yaydoc. If that is implemented, we'll give a link to it, otherwise i think a link to the repository or website is our only option

**Comment by mariobehling** [comment]
@imujjwal96 Please think this through more. One idea is:
Green: Link to deployed website
Gray: Link to fix it on yaydoc
Red: Link to fix it on yaydoc

**Comment by imujjwal96** [comment]
@mariobehling  The status tag cannot have dynamic links. Also, currently there is no way to either see the logs or fix any failed deployment in Yaydoc.
What we can do to achieve something similar to what you mentioned is:

The status tag will link to `https://yaydoc.herokuapp.com/fossasia/susi_server` (example)
At that link, we will check for the build status. if it is `successful`, then we will redirect it to generated docs website. Otherwise, we can show the logs at that url

**Comment by mariobehling** [comment]
@imujjwal96 Ok, let's go with it for now. Cheers.

File: routes/index.js [hunk 3]
```diff
@@ -29,4 +31,37 @@ router.get('/logs/:type/:email/:uniqueId', function (req, res, next) {
...
 });
 
+/* Generate SVG image for build status of repository */
+router.get('/:owner/:reponame.svg', function (req, res, next) {
+  Repository.getBuildStatusByRepositoryName(req.params.owner + '/' + req.params.reponame, function (error, result) {
+    var buildStatus = 'invalid';
+    var width = '94';
+    var color = '#9f9f9f';
+    var x = '70.5';
+    if (result === null) {
```
Comment (sch00lb0y) [inline routes/index.js]: check the error as well.

Comment (imujjwal96) [inline routes/index.js]: What error?

Comment (sch00lb0y) [inline routes/index.js]: db callback error

**Comment by imujjwal96** [comment]
@sch00lb0y  updated

**Pull request merged**